bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -244,12 +278,12 @@ where
 		let inbound_queue_pallet_index = InboundQueuePalletInstance::get();
 
 		let mut instructions = vec![
-			ReceiveTeleportedAsset(total_fee_asset.into()),
-			BuyExecution { fees: asset_hub_fee_asset, weight_limit: Unlimited },
 			DescendOrigin(PalletInstance(inbound_queue_pallet_index).into()),
 			UniversalOrigin(GlobalConsensus(network)),
-			ReserveAssetDeposited(asset.clone().into()),
-			ClearOrigin,
```

Comment:
The reason not use `Teleport` here is for the case when failed at executing the `BuyExecution { fees: asset_hub_fee_asset, weight_limit: Unlimited }` for insufficient fee, the token(weth) is in registry yet so can't be claimed and lost.